### DISCUSSION TITLE
Error in RL model

### DISCUSSION THREAD

**Discussion by AimanKhan1997** [zach401/acnportal#99] [Q&A] [answered]
Hey @sunash , I am currently looking at the RL example in tutorials about[ training_running_rl_agent](https://github.com/caltech-netlab/gym-acnportal/blob/master/tutorials/lessonx_training_running_rl_agent.ipynb).
I have tried running the model but keep receiving error-AttributeError: 'Simulator' object has no attribute 'generate_interface' when I run the interface_generating_function(). I tried looking but couldn't find the solution. 
Kindly let me know how to solve this error.
Moreover, does gym-portal have other models trained for 1e6 iterations just like PPO2 model shown in the example, specifically for DQN and A2C.
Thank you for your time.

**Comment by sunash** [comment, marked as answer]
Good to hear from you! The bug you mentioned is a known issue. I'm working on fixing it completely, but in the meantime, try running the tutorial on the dev branch of gym-acnportal (where I'm currently working on a solution). Not all the tests pass yet, but the tutorial should work.

The package doesn't have any models trained with other algorithms, unfortunately.

**Reply by AimanKhan1997** [reply]
Thanks for letting me know. Also, I wanted to try this model with DCFC. Do Is changing the max_rate and voltage in random_plugin to 114A and 480V enough or do I need to make other changes too? Also in the reward function I wanted to add three more functions of cn.never_charged, energy_cost and demand_charge as mentioned in [infrastructure evaluation example](https://github.com/caltech-netlab/acnportal-experiments/blob/master/examples/1-Infrastructure-Evaluation/1.2-Comparing-Infrastructure-Designs/1.2-comparing-infrastructure-designs.ipynb) in run_experiment function. 
Can you please suggest how I should do it. It would be of great help.
Thanks again for your help.

**Reply by sunash** [reply]
In addition to changing the max rate and voltage in random_plugin, you'll need to alter the network construction in random_plugin. acnsim.sites.simple_acn uses a basic EVSE with max rate 32 A. You'll want to make a new charging network with EVSEs with DCFC voltage and max rate. The simple_acn code uses get_evse_by_type to construct the EVSEs, but you'll probably just want to do that directly (via the EVSE constructor).

In gym-acnportal's CustomSimEnv class, you can input a list of reward functions (functions that take an environment and return a reward) to use in agent evaluation. Since the environment wraps a simulator, you can use as a reward function any function that takes a simulator as input and outputs a number. However, keep in mind that calculating reward functions through the Interface to the wrapped simulator is more practically realizable (using the sim directly may give you information that you don't have in a real setting).

In brief, instead of using the default-sim-env, you'll make your own custom environment using the CustomSimEnv constructor, inputting the reward functions you want to use.

**Reply by aboltakhty** [reply]
Thanks for useful information.
